components/brave_rewards/browser/brave_rewards_data_remover_delegate.cc

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/brave_rewards_data_remover_delegate.h"  // NOLINT
+
+#include <stdint.h>
+
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/barrier_closure.h"
+#include "base/callback.h"
+#include "base/callback_helpers.h"
+#include "base/task/post_task.h"
+#include "brave/components/brave_rewards/browser/rewards_service.h"
+#include "brave/components/brave_rewards/browser/rewards_service_factory.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/io_thread.h"
+#include "chrome/browser/profiles/profile.h"
+#include "brave/common/pref_names.h"
+#include "chrome/common/url_constants.h"
+#include "components/prefs/pref_service.h"
+#include "content/public/browser/browser_task_traits.h"
+#include "content/public/browser/browser_thread.h"
+
+using content::BrowserContext;
+using content::BrowserThread;
+using brave_rewards::RewardsService;
+using brave_rewards::RewardsServiceFactory;
+
+BraveRewardsDataRemoverDelegate::BraveRewardsDataRemoverDelegate(
+    BrowserContext* browser_context)
+    : ChromeBrowsingDataRemoverDelegate(browser_context),
+      profile_(Profile::FromBrowserContext(browser_context)) {
+}
+
+BraveRewardsDataRemoverDelegate::~BraveRewardsDataRemoverDelegate() {}
+
+void BraveRewardsDataRemoverDelegate::RemoveEmbedderData(
+    const base::Time& delete_begin,
+    const base::Time& delete_end,
+    int remove_mask,
+    const content::BrowsingDataFilterBuilder& filter_builder,
+    int origin_type_mask,
+    base::OnceClosure callback) {
+
+  ChromeBrowsingDataRemoverDelegate::RemoveEmbedderData(
```

Comment:
this is going to call the callback before we've even started to remove the rewards data. `base::DoNothing` below should instead call `ChromeBrowsingDataRemoverDelegate::RemoveEmbedderData`